planner/core/point_get_plan.go

```diff
@@ -1137,3 +1117,42 @@ func findPartitionIdx(idxInfo *model.IndexInfo, pos int, pairs []nameValuePair)
 	}
 	return 0
 }
+
+// getPartitionColumnPos gets the partition column's position in the index.
+func getPartitionColumnPos(idx *model.IndexInfo, partitionColName *ast.ColumnName) int {
+	if partitionColName == nil {
+		return 0
+	}
+	for i, idxCol := range idx.Columns {
+		if partitionColName.Name.L == idxCol.Name.L {
```

Comment:
I'm not sure could the name works well in the `col as name` alias pattern.
@SunRunAway